cpp/src/merge/merge.cu

```diff
@@ -45,8 +53,49 @@
 
 namespace cudf {
 namespace detail {
+
 namespace {
 
+template <bool has_nulls>
+struct row_lexicographic_tagged_comparator {
+  row_lexicographic_tagged_comparator(table_device_view const lhs,
+                                      table_device_view const rhs,
+                                      device_span<order const> const column_order,
+                                      device_span<null_order const> const null_precedence)
+    : _lhs{lhs}, _rhs{rhs}, _column_order{column_order}, _null_precedence{null_precedence}
+  {
+  }
+
+  __device__ bool operator()(index_type lhs_tagged_index,
+                             index_type rhs_tagged_index) const noexcept
+  {
+    auto const [l_side, l_indx] = lhs_tagged_index;
+    auto const [r_side, r_indx] = rhs_tagged_index;
+
+    table_device_view const* ptr_left_dview{l_side == side::LEFT ? &_lhs : &_rhs};
+    table_device_view const* ptr_right_dview{r_side == side::LEFT ? &_lhs : &_rhs};
+    auto comparator = [&]() {
+      if (has_nulls) {
```

Comment:
Silly question, for my edumacation: Given that `has_nulls` is `constexpr`, is there any value at all making this `if` an `if constexpr`? My guess is no, but I'd like to confirm.

By my understanding:
1. `if constexpr` is to be used where either the `if` or `else` branch might not compile, depending on the value of `has_nulls`.
2. The advantage of "optimizing away the `if`-check" is independent of whether this is a `constexpr` check.

Just confirming my understanding here.

Reply:
Your understanding is spot-on, actually, thanks for pointing this out! This should be `if constexpr`. Not only can we reduce compilation time this way, we also make the device function less bloated by not having to instantiate two versions.